doc/kernel.txt

```diff
@@ -0,0 +1,5 @@
+Kernel protocol
```

Comment:
Maybe an `.md` file?

Can you describe in a single sentence where this is used in the library?